homeassistant/components/homekit_controller/button.py

```diff
@@ -46,17 +46,42 @@ class HomeKitButtonEntityDescription(ButtonEntityDescription):
 }
 
 
+@dataclass
+class HomeKitEcobeeButtonEntityDescription(ButtonEntityDescription):
+    """Describes Homekit button."""
+
+    write_values: list[int | str | None] = field(default_factory=list)
+
+
+ECOBEE_BUTTON_ENTITIES: dict[str, HomeKitEcobeeButtonEntityDescription] = {
+    CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD: HomeKitEcobeeButtonEntityDescription(
+        key=CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD,
+        name="Clear Hold",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=ENTITY_CATEGORY_CONFIG,
+        write_values=[False, True],
+    ),
+}
+
+
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up Homekit buttons."""
     hkid = config_entry.data["AccessoryPairingID"]
     conn = hass.data[KNOWN_DEVICES][hkid]
 
     @callback
     def async_add_characteristic(char: Characteristic):
-        if not (description := BUTTON_ENTITIES.get(char.type)):
-            return False
+        entities: list[HomeKitButton | HomeKitEcobeeButton] = []
```

Comment:
You could just use a super class here (`ButtonEntity`). Although if the type signature on `async_setup_entry` was right (anywhere in homekit_controller) I don't think the annotation would be worth adding.